app/controllers/admin/reports_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Admin::ReportsController < Admin::BaseController
+  def index
+  end
+
+  def deals
+    @deals = Deal.deal_revenue
+    flash[:notice] = 'No customers found' if @deals.empty?
+  end
+
+  def customers
+    if params[:from] && params[:to]
+      @customers = User.top_spending_customers(params[:from], params[:to])
+      flash[:notice] = 'No customers found' if @customers.empty?
+    end
```

Comment:
What if one of params[:from] or params[:to] is absent or both are absent?

Reply:
it will pass empty string which will give SQL error, so to handle this we need to check if its present or not and also set default values in scope.